pymeasure/instruments/velleman/velleman_k8090.py

```diff
@@ -0,0 +1,249 @@
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import serial
+
+from pymeasure.instruments import Instrument
+
+
+def _parse_channels(channels) -> str:
+    """Convert array of channel numbers into mask if needed."""
+    if isinstance(channels, list):
+        mask = 0x00
+        for ch in channels:
+            mask |= 1 << (ch - 1)
+    else:
+        mask = channels
+
+    return hex(mask)
+
+
+def _get_process_status(items):
+    """Process the result of a 0x51 status message."""
+    if len(items) < 4 or items[0] != 0x51:
+        return None
+
+    return _ints_to_bool_lists(items[1:])
+
+
+def _ints_to_bool_lists(numbers):
+    """Convert 8-bit integers into lists of booleans.
+
+    Also works for a single given integer.
+
+    The least significant bit (#0) will be first
+    """
+    if isinstance(numbers, int):
+        numbers = [numbers]
+
+    bool_lists = [[bool(num & (1 << n)) for n in range(8)] for num in numbers]
+
+    if len(bool_lists) == 1:
+        return bool_lists[0]
+
+    return bool_lists
+
+
+class VellemanK8090(Instrument):
+    """For usage with the K8090 relay board, by Velleman.
+
+    The communication is done by serial USB.
+
+    View the "K8090/VM8090 PROTOCOL MANUAL" for the serial command instructions.
+    """
+
+    BYTE_STX = 0x04
+    BYTE_ETX = 0x0F
+
+    version = Instrument.measurement(
+        "0x71",
+        """
+        Get firmware version, as (year - 2000, week).
+        E.g. ``(10, 1)``
+        """,
+        cast=int,
+        get_process=lambda v: (v[2], v[3]) if len(v) > 3 and v[0] == 0x71 else None,
+    )
+
+    status = Instrument.measurement(
+        "0x18",
+        """
+        Get current relay status.
+        The reply has a different command byte than the request.
+        
+        Three items (lists of 8 bools) are returned:
+        - Previous state: the state of each relay before this event
+        - Current state: the state of each relay now
+        - Timer state: the state of each relay timer
+        """,
+        cast=int,
+        get_process=_get_process_status,
+    )
+
+    switch_on = Instrument.setting(
+        "0x11,%s",
+        """"
+        Switch on a set of channels.
+        Other channels are unaffected.
+        Pass either a list or set of channel numbers (starting at 1), or pass a bitmask
+        """,
+        set_process=_parse_channels,
+    )
+
+    switch_off = Instrument.setting(
+        "0x12,%s",
+        """"
+        Switch off a set of channels. See :prop:`switch_on`.
+        """,
+        set_process=_parse_channels,
+    )
+
+    id = None  # No identification available
+
+    def __init__(
+        self, adapter, name="Velleman K8090", baud_rate=19200, timeout=1000, **kwargs
+    ):
+        super().__init__(
+            adapter, name=name, baud_rate=baud_rate, timeout=timeout, **kwargs
+        )
```

Comment:
Please move the init to the beginning of the file and please make the baud_rate a "specific parameter" for serial connections (that facilitates other adapters):
```suggestion
    def __init__(
        self, adapter, name="Velleman K8090", baud_rate=19200, timeout=1000, **kwargs
    ):
        super().__init__(
            adapter, name=name, asrl={'baud_rate': baud_rate}, timeout=timeout, **kwargs
        )
```

Is the baud_rate settable? If not, you could replace it with the number 19200.

Reply:
Ah, I didn't know about the `asrl` argument. Changed and I hardcoded the 19200, because the K8090 does not support a different rate anyway.

Does the instrument now insert the baudrate into the adapter that's passed, regardless of the adapter constructor parameters?